File: code/netcore/src/Org.OpenAPITools/Model/GezagsverhoudingInOnderzoek.cs
/*
 * Bevragen Ingeschreven Personen RvIG
 *
 * API voor het bevragen van ingeschreven personen uit de basisregistratie personen (BRP), inclusief de registratie niet-ingezeten (RNI). Met deze API kun je personen zoeken en actuele gegevens over personen, kinderen, partners en ouders raadplegen.  Gegevens die er niet zijn of niet actueel zijn krijg je niet terug. Heeft een persoon bijvoorbeeld geen geldige nationaliteit, en alleen een beëindigd partnerschap, dan krijg je geen gegevens over nationaliteit en partner.  Zie de [Functionele documentatie](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/tree/v1.2.0/features) voor nadere toelichting.  De API zal door RvIG worden aangeboden. Daarin wordt een aantal afgeleide gegevens niet opgenomen. Daarom bevat deze specificatie minder gegevens dan de specificatie op [BRP-bevragen](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/master/specificatie/genereervariant/openapi.yaml). 
 *
 * The version of the OpenAPI document: 1.2.0
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Org.OpenAPITools.Client.OpenAPIDateConverter;

namespace Org.OpenAPITools.Model
{
    /// <summary>
    /// Geeft aan welke gegevens van de gezagsverhouding in onderzoek zijn. Zie de [functionele specificaties](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.1.0/features/in_onderzoek.feature) 
    /// </summary>
    [DataContract(Name = "GezagsverhoudingInOnderzoek")]
    public partial class GezagsverhoudingInOnderzoek : IEquatable<GezagsverhoudingInOnderzoek>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="GezagsverhoudingInOnderzoek" /> class.
        /// </summary>
        /// <param name="indicatieCurateleRegister">indicatieCurateleRegister.</param>
        /// <param name="indicatieGezagMinderjarige">indicatieGezagMinderjarige.</param>
        /// <param name="datumIngangOnderzoek">datumIngangOnderzoek.</param>
        public GezagsverhoudingInOnderzoek(bool indicatieCurateleRegister = default(bool), bool indicatieGezagMinderjarige = default(bool), DatumOnvolledig datumIngangOnderzoek = default(DatumOnvolledig))
        {
            this.IndicatieCurateleRegister = indicatieCurateleRegister;
            this.IndicatieGezagMinderjarige = indicatieGezagMinderjarige;
            this.DatumIngangOnderzoek = datumIngangOnderzoek;
        }

        /// <summary>
        /// Gets or Sets IndicatieCurateleRegister
        /// </summary>
        [DataMember(Name = "indicatieCurateleRegister", EmitDefaultValue = false)]
        public bool IndicatieCurateleRegister { get; set; }

        /// <summary>
        /// Gets or Sets IndicatieGezagMinderjarige
        /// </summary>
        [DataMember(Name = "indicatieGezagMinderjarige", EmitDefaultValue = false)]
        public bool IndicatieGezagMinderjarige { get; set; }

        /// <summary>
        /// Gets or Sets DatumIngangOnderzoek
        /// </summary>
        [DataMember(Name = "datumIngangOnderzoek", EmitDefaultValue = false)]
        public DatumOnvolledig DatumIngangOnderzoek { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class GezagsverhoudingInOnderzoek {\n");
            sb.Append("  IndicatieCurateleRegister: ").Append(IndicatieCurateleRegister).Append("\n");
            sb.Append("  IndicatieGezagMinderjarige: ").Append(IndicatieGezagMinderjarige).Append("\n");
            sb.Append("  DatumIngangOnderzoek: ").Append(DatumIngangOnderzoek).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as GezagsverhoudingInOnderzoek);
        }

        /// <summary>
        /// Returns true if GezagsverhoudingInOnderzoek instances are equal
        /// </summary>
        /// <param name="input">Instance of GezagsverhoudingInOnderzoek to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(GezagsverhoudingInOnderzoek input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.IndicatieCurateleRegister == input.IndicatieCurateleRegister ||
                    this.IndicatieCurateleRegister.Equals(input.IndicatieCurateleRegister)
                ) && 
                (
                    this.IndicatieGezagMinderjarige == input.IndicatieGezagMinderjarige ||
                    this.IndicatieGezagMinderjarige.Equals(input.IndicatieGezagMinderjarige)
                ) && 
                (
                    this.DatumIngangOnderzoek == input.DatumIngangOnderzoek ||
                    (this.DatumIngangOnderzoek != null &&
                    this.DatumIngangOnderzoek.Equals(input.DatumIngangOnderzoek))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                hashCode = hashCode * 59 + this.IndicatieCurateleRegister.GetHashCode();
                hashCode = hashCode * 59 + this.IndicatieGezagMinderjarige.GetHashCode();
                if (this.DatumIngangOnderzoek != null)
                    hashCode = hashCode * 59 + this.DatumIngangOnderzoek.GetHashCode();
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> IValidatableObject.Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}

File: code/java/src/main/java/org/openapitools/client/model/GezagsverhoudingInOnderzoek.java
/*
 * Bevragen Ingeschreven Personen RvIG
 * API voor het bevragen van ingeschreven personen uit de basisregistratie personen (BRP), inclusief de registratie niet-ingezeten (RNI). Met deze API kun je personen zoeken en actuele gegevens over personen, kinderen, partners en ouders raadplegen.  Gegevens die er niet zijn of niet actueel zijn krijg je niet terug. Heeft een persoon bijvoorbeeld geen geldige nationaliteit, en alleen een beëindigd partnerschap, dan krijg je geen gegevens over nationaliteit en partner.  Zie de [Functionele documentatie](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/tree/v1.2.0/features) voor nadere toelichting.  De API zal door RvIG worden aangeboden. Daarin wordt een aantal afgeleide gegevens niet opgenomen. Daarom bevat deze specificatie minder gegevens dan de specificatie op [BRP-bevragen](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/master/specificatie/genereervariant/openapi.yaml). 
 *
 * The version of the OpenAPI document: 1.2.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package org.openapitools.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.io.IOException;
import org.openapitools.client.model.DatumOnvolledig;

/**
 * Geeft aan welke gegevens van de gezagsverhouding in onderzoek zijn. Zie de [functionele specificaties](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.1.0/features/in_onderzoek.feature) 
 */
@ApiModel(description = "Geeft aan welke gegevens van de gezagsverhouding in onderzoek zijn. Zie de [functionele specificaties](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.1.0/features/in_onderzoek.feature) ")
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", date = "2021-02-22T08:00:54.409Z[Etc/UTC]")
public class GezagsverhoudingInOnderzoek {
  public static final String SERIALIZED_NAME_INDICATIE_CURATELE_REGISTER = "indicatieCurateleRegister";
  @SerializedName(SERIALIZED_NAME_INDICATIE_CURATELE_REGISTER)
  private Boolean indicatieCurateleRegister;

  public static final String SERIALIZED_NAME_INDICATIE_GEZAG_MINDERJARIGE = "indicatieGezagMinderjarige";
  @SerializedName(SERIALIZED_NAME_INDICATIE_GEZAG_MINDERJARIGE)
  private Boolean indicatieGezagMinderjarige;

  public static final String SERIALIZED_NAME_DATUM_INGANG_ONDERZOEK = "datumIngangOnderzoek";
  @SerializedName(SERIALIZED_NAME_DATUM_INGANG_ONDERZOEK)
  private DatumOnvolledig datumIngangOnderzoek;


  public GezagsverhoudingInOnderzoek indicatieCurateleRegister(Boolean indicatieCurateleRegister) {
    
    this.indicatieCurateleRegister = indicatieCurateleRegister;
    return this;
  }

   /**
   * Get indicatieCurateleRegister
   * @return indicatieCurateleRegister
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "")

  public Boolean getIndicatieCurateleRegister() {
    return indicatieCurateleRegister;
  }


  public void setIndicatieCurateleRegister(Boolean indicatieCurateleRegister) {
    this.indicatieCurateleRegister = indicatieCurateleRegister;
  }


  public GezagsverhoudingInOnderzoek indicatieGezagMinderjarige(Boolean indicatieGezagMinderjarige) {
    
    this.indicatieGezagMinderjarige = indicatieGezagMinderjarige;
    return this;
  }

   /**
   * Get indicatieGezagMinderjarige
   * @return indicatieGezagMinderjarige
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "")

  public Boolean getIndicatieGezagMinderjarige() {
    return indicatieGezagMinderjarige;
  }


  public void setIndicatieGezagMinderjarige(Boolean indicatieGezagMinderjarige) {
    this.indicatieGezagMinderjarige = indicatieGezagMinderjarige;
  }


  public GezagsverhoudingInOnderzoek datumIngangOnderzoek(DatumOnvolledig datumIngangOnderzoek) {
    
    this.datumIngangOnderzoek = datumIngangOnderzoek;
    return this;
  }

   /**
   * Get datumIngangOnderzoek
   * @return datumIngangOnderzoek
  **/
  @javax.annotation.Nullable
  @ApiModelProperty(value = "")

  public DatumOnvolledig getDatumIngangOnderzoek() {
    return datumIngangOnderzoek;
  }


  public void setDatumIngangOnderzoek(DatumOnvolledig datumIngangOnderzoek) {
    this.datumIngangOnderzoek = datumIngangOnderzoek;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    GezagsverhoudingInOnderzoek gezagsverhoudingInOnderzoek = (GezagsverhoudingInOnderzoek) o;
    return Objects.equals(this.indicatieCurateleRegister, gezagsverhoudingInOnderzoek.indicatieCurateleRegister) &&
        Objects.equals(this.indicatieGezagMinderjarige, gezagsverhoudingInOnderzoek.indicatieGezagMinderjarige) &&
        Objects.equals(this.datumIngangOnderzoek, gezagsverhoudingInOnderzoek.datumIngangOnderzoek);
  }

  @Override
  public int hashCode() {
    return Objects.hash(indicatieCurateleRegister, indicatieGezagMinderjarige, datumIngangOnderzoek);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class GezagsverhoudingInOnderzoek {\n");
    sb.append("    indicatieCurateleRegister: ").append(toIndentedString(indicatieCurateleRegister)).append("\n");
    sb.append("    indicatieGezagMinderjarige: ").append(toIndentedString(indicatieGezagMinderjarige)).append("\n");
    sb.append("    datumIngangOnderzoek: ").append(toIndentedString(datumIngangOnderzoek)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

